feat: update terraform test schema for v1.11

MorrisWitthein opened the pull request:
Fixes: #449

MorrisWitthein committed:
feat: update terraform test schema for v1.11

anubhav-goel commented on internal/schema/tests/1.11/root.go:
```diff
@@ -0,0 +1,22 @@
+func TestSchema(v *version.Version) *schema.BodySchema {
+	bs := v1_9_test.TestSchema(v)
+
+	bs.Blocks["run"].Body.Blocks["state_key"] = stateKeyBlockSchema()
```
`state_key` is actually an attribute rather than a nested block. 
So it should be added to Attributes instead of Blocks. For a nice reference on how the codebase handles this pattern, take a look at https://github.com/hashicorp/terraform-schema/pull/492

anubhav-goel commented on internal/schema/tests/1.11/state_key_block.go:
```diff
@@ -0,0 +1,25 @@
+			Attributes: map[string]*schema.AttributeSchema{
+				"state_key": {
+					Constraint:  schema.AnyExpression{OfType: cty.String},
+					IsRequired:  false,
```
Instead of isRequired: false, it's a good practice to use isOptional: true

anubhav-goel commented on internal/schema/tests/1.11/root.go:
```diff
@@ -0,0 +1,22 @@
+	bs := v1_9_test.TestSchema(v)
+
+	bs.Blocks["run"].Body.Blocks["state_key"] = stateKeyBlockSchema()
+	bs.Blocks["mock_provider"].Body.Blocks["override_during"] = overrideDuringBlockSchema()
```
Same comment as above, `override_during` is also an attribute rather than a nested block. 

Also, `override_during` is only added to mock_provider, it should be added in other places as well. Please refer to the documentation for the same: https://developer.hashicorp.com/terraform/language/tests/mocking

anubhav-goel reviewed (changes requested):
Hi @MorrisWitthein,
Thank you so much for taking the time to put this together! 🎉 It's great to see a community contribution tackling [#449](https://github.com/hashicorp/terraform-schema/issues/449). 
I've left a few comments below, mostly around how `state_key` and `override_during` are modeled in the schema.

MorrisWitthein committed:
fix: convert state_key & override_during from BlockSchema to
AttributeSchema and add missing occurences of override_during attribute

MorrisWitthein commented:
Hey @anubhav-goel,
thanks a lot for your feedback. I converted both of them to attributes and added the missing definitions for the `override_during` attribute.

I added a central definition for the `override_during` attribute in the `root.go` file. Let me know if that is fine, or breaking with your current patterns 🙂

anubhav-goel commented on internal/schema/tests/1.11/root.go:
```diff
@@ -0,0 +1,45 @@
+
+func overrideDuringAttributeSchema() *schema.AttributeSchema {
+	return &schema.AttributeSchema{
+		Description: lang.PlainText("Allows overriding the point in time where terraform generates data"),
```
`lang.Markdown` could be used instead

anubhav-goel commented on internal/schema/tests/1.11/root.go:
```diff
@@ -0,0 +1,45 @@
+	return bs
+}
+
+func overrideDuringAttributeSchema() *schema.AttributeSchema {
```
overrideDuringAttributeSchema() can be moved to a shared file (e.g. internal/schema/tests/1.11/override_during_attribute.go) to keep root.go/patch files focused.

anubhav-goel commented on internal/schema/tests/1.11/root.go:
```diff
@@ -0,0 +1,45 @@
+	bs.Blocks["run"] = patchRunBlockSchema(bs.Blocks["run"])
+	bs.Blocks["mock_provider"] = patchMockProviderBlockSchema(bs.Blocks["mock_provider"])
+	bs.Blocks["mock_resource"] = patchMockResourceBlockSchema(bs.Blocks["mock_resource"])
+	bs.Blocks["mock_data"] = patchMockDataBlockSchema(bs.Blocks["mock_data"])
```
`mock_resource` and `mock_data` are not root blocks, they are nested blocks inside mock_provider. Adding them at the root schema level could cause terraform-ls to mishandle the schema and potentially crash the server.

anubhav-goel reviewed (commented):
Thanks for the update!! Most of the changes look good overall.

However, the current changes will likely crash the terraform-ls server (since mock_resource and mock_data are not root blocks, but are being modeled at the root schema level).

Hence, I’d recommend linking terraform-schema, terraform-ls, and vscode-terraform and validating end-to-end by running the full setup to confirm it achieves the desired result. If possible, sharing a short video of the behavior working in VS Code would also be helpful.

MorrisWitthein committed:
fix: extract override_during attribute to own file and move mock_resource/mock_data patching into mock_provider_block

MorrisWitthein committed:
fix: use Markdown instead of PlainText for override_during description

MorrisWitthein commented:
Demo:
https://github.com/user-attachments/assets/c7785a98-c0b6-4a03-b9ff-15f726fbd789

MorrisWitthein commented:
Thanks again for all your Feedback!
I update everything accordingly and tested the new schema.
